source/GameInterface/Services/Towns/Patches/DailyTickTownPatch.cs

```diff
@@ -0,0 +1,51 @@
+﻿using HarmonyLib;
+using System.Collections.Generic;
+using System.Reflection.Emit;
+using System.Reflection;
+using System;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.Encounters;
+using System.Linq;
+using static HarmonyLib.Code;
+using Common.Messaging;
+using GameInterface.Services.Fiefs.Messages;
+using Newtonsoft.Json.Linq;
+using Autofac.Core.Activators;
+using GameInterface;
+
+[HarmonyPatch(typeof(Town), "DailyTick")]
+public static class TownDailyTickPatch
+{
+    // Get the PropertyInfo for the FoodStocks property
+    private static PropertyInfo foodStocksProperty = typeof(Fief).GetProperty("FoodStocks");
+
+    // Get the setter method for the FoodStocks property
+    private static MethodInfo foodStocksSetter = foodStocksProperty.GetSetMethod(true);
+
+    [HarmonyPatch(typeof(Town), "DailyTick")]
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instrs)
+    {
+        List<CodeInstruction> instructions = instrs.ToList();
+        int setterLastIdx = instructions.FindLastIndex(i => i.opcode == Call.opcode && i.operand as MethodInfo == foodStocksSetter);
+        if (setterLastIdx == -1) return instrs;
+
+        instructions.InsertRange(setterLastIdx, new CodeInstruction[]
+        {
+            new CodeInstruction(OpCodes.Ldarg_0),
+            new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(TownDailyTickPatch), "LogFoodStock"))
+        });
+        
+        return instructions;
```

Comment:
Currently this will only patch the last FoodStocks set

This also does not block setting on the client

Below will replace the call with your method

The stack will also contain the new food stocks value so you will have to set it yourself for more control

```suggestion
        foreach (var instriction in instrs)
{
    if (i.opcode == Call.opcode && i.operand as MethodInfo == foodStocksSetter)
    {
        yield return new CodeInstruction(OpCodes.Ldarg_0);
        yield return new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(TownDailyTickPatch), "LogFoodStock"));
        continue;
    }

    yield return instriction
}
```

Reply:
When I remplace with this code, I have this crash :

![image](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/assets/23268707/14cb0acf-1232-4d38-95f7-97ff80db47a5)

Reply:
The crash seem to come from the yield return

Reply:
So I was able to make it work without crash by removing the new code insruction(OpCodes.Ldarg0)
the crash occurs because we add two new code instruction for one instruction replace
but now I have this as a result : 
![image](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/assets/23268707/4c773c5c-5525-4969-8725-81ee47f866a2)
the fief instance return here for exemple is a list of "village", maybe because 3 villages have the same food stock in same time ?

Reply:
Finally make it works !
```
    foreach (var instruction in instrs)
    {
        if (instruction.opcode == Call.opcode && instruction.operand as MethodInfo == foodStocksSetter)
        {
	    yield return new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(TownDailyTickPatch), "LogFoodStock"));
            //yield return new CodeInstruction(OpCodes.Ldarg_0);
            continue;
        }
        yield return instruction;
    }
    
}

public static void LogFoodStock(Fief fief, float value)```

so I remove the Ldarg_0 and change the order of the parameter that take LogFoodStock, the problem was there